src/python/pants/backend/python/goals/run_helper.py

```diff
@@ -120,14 +120,13 @@ async def _create_python_source_run_request(
         *chrooted_source_roots,
     ]
     extra_env = {
-        **pex_env.in_workspace().environment_dict(python_configured=pex.python is not None),
-        "PEX_PATH": _in_chroot(local_dists.pex.name),
+        **complete_pex_environment.environment_dict(python_configured=True),
```

Comment:
Why is it true that pex.python is not None always here?

Reply:
A `VenvPex` has a mandatory `python` attribute

Reply:
I just did a `is not None` and `mypy` didn't complain. Will push that for clarity.